core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -68,6 +68,9 @@ private[spark] class Executor(
 
   // Make sure the local hostname we report matches the cluster scheduler's name for this host
   Utils.setCustomHostname(executorHostname)
+  
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
```

Comment:
since you can have multiple executors on one host, we need host & port to uniquely identify each executor, right?  Alternatively we could just use the `executorID`, and in the UI show `executorId` and `host:port`